app/models/storage_service.rb

```diff
@@ -17,4 +17,107 @@ def self.class_by_ems(ext_management_system)
     # side, that would return correct class for each provider
     ext_management_system && ext_management_system.class::StorageService
   end
+
+
+  def self.create_storage_service_queue(userid, ext_management_system, options = {})
```

Comment:
The rest of this doesn't look related to allowing the addition of standalone storage managers, was this intended to be for another PR?